Utils/FaviconFetcher.swift

```diff
@@ -93,73 +94,84 @@ public class FaviconFetcher : NSObject, NSXMLParserDelegate {
-                        iconType = .Guess
-                    }
+            if let url = reloadUrl {
+              return self.parseHTMLForFavicons(url)
+            }
 
-                    if let type = iconType where !bestType.isPreferredTo(type),
-                        let iconUrl = NSURL(string: href, relativeToURL: url) {
-                            let icon = Favicon(url: iconUrl.absoluteString, date: NSDate(), type: type)
-                            // If we already have a list of Favicons going already, then add it…
-                            if (type == bestType) {
-                                icons.append(icon)
-                            } else {
-                                // otherwise, this is the first in a new best yet type.
-                                icons = [icon]
-                                bestType = type
-                            }
+            //map all the icon types to an array of Favicons
+            let headElement = element["html"]["head"]["link"]
+            let relAttributes = ["shortcut icon","icon","apple-touch-icon","apple-touch-icon-precomposed"]
+            let iconsMapped = relAttributes.flatMap({ (relIcon:String) -> Favicon? in
+                do {
+                    guard let iconHref = try headElement.withAttr("rel", relIcon).element?.attributes["href"] else {
+                        return nil
+                    }
+                    if let iconUrl = NSURL(string: iconHref, relativeToURL: url) {
+                        let type = iconHref.endsWith(".ico") ? IconType.Guess : self.iconTypeForLink(relIcon)
+                        let icon = Favicon(url: iconUrl.absoluteString, date: NSDate(), type: type)
+                        return icon
                     }
+                } catch {
+                    return nil
                 }
+                return nil
+            })
 
-                // If we haven't got any options icons, then use the default at the root of the domain.
-                if let url = NSURL(string: "/favicon.ico", relativeToURL: url) where icons.isEmpty {
-                    let icon = Favicon(url: url.absoluteString, date: NSDate(), type: .Guess)
-                    icons = [icon]
-                }
+            //sort that array such that the first Item is the perferred type
+            icons = iconsMapped.sort({ (a:Favicon, b:Favicon) -> Bool in
+              return a.type.isPreferredTo(b.type)
+            })
+
+            // If we haven't got any options icons, then use the default at the root of the domain.
+            if let url = NSURL(string: "/favicon.ico", relativeToURL: url) where icons.isEmpty {
+                let icon = Favicon(url: url.absoluteString, date: NSDate(), type: .Guess)
+                icons.append(icon)
             }
+            //only return the icon with the highest resolution
+            if let bestIcon = icons.first {
+                icons = [bestIcon]
```

Comment:
I only return one Icon in the end. I'm not sure if this is okay. But it made sense to only return the best Image.